composer.json

```diff
@@ -27,28 +27,33 @@
     "config": {
         "sort-packages": true
     },
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/webimpress/zend-expressive-router.git"
+        }
+    ],
     "require": {
-        "php": "^5.6 || ^7.0",
+        "php": "^7.1",
         "fig/http-message-util": "^1.1.2",
-        "http-interop/http-middleware": "^0.4.1",
+        "http-interop/http-server-middleware": "^1.0.1",
         "psr/container": "^1.0",
         "psr/http-message": "^1.0.1",
-        "webimpress/http-middleware-compatibility": "^0.1.1",
         "zendframework/zend-diactoros": "^1.3.10",
-        "zendframework/zend-expressive-router": "^2.2",
+        "zendframework/zend-expressive-router": "dev-feature/psr-15 as 2.2",
         "zendframework/zend-expressive-template": "^1.0.4",
-        "zendframework/zend-stratigility": "^2.1"
+        "zendframework/zend-stratigility": "dev-release-3.0.0"
```

Comment:
Maybe try one of the following constraints:

- `^3.0.0@dev`
- `^3.0.0@dev as 3.0.0alpha1`
- `^3.0.0-dev`
- `^3.0.0-dev as 3.0.0alpha1`

My understanding is that we should use semver constraints instead of the branch name if an alias exists; I just can't recall off the top of my head what that should look like. Experiment! :grin:

Reply:
Looks like `^3.0.0@dev` and `^3.0.0-dev` are the same, when I try with alias I got error:
> Problem 1
    - The requested package zendframework/zend-stratigility ^3.0.0@dev as 3.0.0alpha1 exists as zendframework/zend-stratigility[1.0.0, 1.0.1, 1.0.2, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, dev-develop, 2.2.x-dev, dev-master, 2.1.x-dev, dev-release-3.0.0, 3.0.x-dev] but these are rejected by your constraint.